xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -587,8 +587,12 @@ bool CGUIWindowPVRBase::EditTimer(CFileItem *item)
     return false;
   }
 
-  if (ShowTimerSettings(timer) && !timer->GetTimerType()->IsReadOnly())
-    return g_PVRTimers->UpdateTimer(timer);
+  // clone the timer.
+  const CPVRTimerInfoTagPtr newTimer(new CPVRTimerInfoTag);
+  newTimer->UpdateEntry(timer);
+
+  if (ShowTimerSettings(newTimer) && !timer->GetTimerType()->IsReadOnly())
+    return g_PVRTimers->UpdateTimer(newTimer);
```

Comment:
Sounds complicated but sane. Does this mean timee updates will only be reflected in the UI once the backend has processes the request and triggered Kodi to update timers?